.pre-commit-config.yaml

```diff
@@ -56,12 +56,37 @@ repos:
     -   id: incorrect-sphinx-directives
         name: Check for incorrect Sphinx directives
         language: pygrep
-        entry: >-
-            \.\. (autosummary|contents|currentmodule|deprecated
-            |function|image|important|include|ipython|literalinclude
-            |math|module|note|raw|seealso|toctree|versionadded
-            |versionchanged|warning):[^:]
+        entry: |
+            (?x)\.\.\ (
+                autosummary|contents|currentmodule|deprecated|
+                function|image|important|include|ipython|literalinclude|
+                math|module|note|raw|seealso|toctree|versionadded|
+                versionchanged|warning
+            ):[^:]
         files: \.(py|pyx|rst)$
+    -   id: non-standard-imports
+        name: Check for non-standard imports
+        language: pygrep
+        entry: |
+            (?x)^(
+                from\ pandas\.core\.common\ import|
+                from\ pandas\.core\ import\ common|
+                from\ collections\.abc\ import|
+                from\ numpy\ import\ nan
+            )
+        types: [python]
+    -   id: non-standard-imports-in-tests
```

Comment:
can you add some comments here like we have in the code_checks.sh; e.g. what exactly some of these mean (the name is fine, but need a bit more), e.g. what are we testing for here

Reply:
@jreback sure, I've added some comments (they're ignored by verbose regexes anyway)